lib/syskit/network_generation/engine.rb

```diff
@@ -311,16 +311,6 @@ def find_selected_device_in_hierarchy(argument_name, leaf_task, requirements)
                 _, model, _ = leaf_task.requirements.resolved_dependency_injection.selection_for(nil, requirements)
                 if model && dev = model.arguments[argument_name]
                     return dev
-                else
-                    devices = Set.new
```

Comment:
> This prevents automatic selection of devices. This indeed has an influence on syskit's behavior as compared to before the PR.

And because of this, I would much prefer having it in a different PR.